app/models/like.rb

```diff
@@ -19,6 +19,10 @@ def relayable_options
 
   after_destroy do
     self.parent.update_likes_counter
+    participation = Participation.where({ 
+      target_id: self.target_id, author_id: self.author_id
+    }).first()
+    participation.update!(count: participation.count.pred) if participation.present?
```

Comment:
That code is redundant, let's extract it somewhere, `Participation::add_reference` or something seems reasonable. That would also be a good place to check whether the counter reached 0 then and it would simplify the unit tests for Comment & Like, you just check whether the method was called, functionality of it only needs to be tested once then in the unit tests for Participation.